tests/Doctrine/Tests/ORM/EntityManagerTest.php

```diff
@@ -288,4 +291,34 @@ public function testGetReferenceRetrievesReferencesWithGivenProxiesAsIdentifiers
         self::assertTrue($this->em->contains($simpleIdReferenceB));
         self::assertTrue($this->em->contains($nestedIdReference));
     }
+
+    public function testCallingFlushInsideFlush()
+    {
+        $listener = new class($this)
+        {
+            /** @var OrmTestCase */
+            private $testCase;
+
+            /** @var bool This flag is used to prevent infinity loop in case of regression */
+            private $alreadyInPreFlush = false;
+
+            public function __construct(OrmTestCase $testCase)
+            {
+                $this->testCase = $testCase;
+            }
+
+            public function preFlush(PreFlushEventArgs $eventArgs)
+            {
+                if ($this->alreadyInPreFlush) {
+                    $this->testCase->fail('Flush inside flush called. It should not happened. Fail test to prevent infinity loop');
```

Comment:
This line is too long, let's use linebreaks.
```suggestion
                    $this->testCase->fail(
                        'Flush inside flush called. It should not happened. Fail test to prevent infinity loop'
                    );
```